backups/config.txt
#!/usr/bin/env python
#-*- coding: utf-8 -*-
##---------------------------------------------------------------------------------------
#  Configuration and variables
#
##---------------------------------------------------------------------------------------

import styles
import os

# set this to True to get lots of messaging
debug = True

# set this to true if you are manually editing your main file
# then back to False when you're satisfied it loads properly
# game will NOT save if this is True
manual_edit_mode = False

# this controls how often full zip backups are made
# options are 'app_exit' and 'app_start', and you can use both
backup_behavior = ['app_exit']
# backup_limit should be 0 (for unlimited) or a positive number
# if set to a negative, no backups will be made
backup_limit = 7

#------------------------------------------------------------------------
# Changes to the following are stored in save games
#------------------------------------------------------------------------

general = dict(
    pretitle = '------------------------------------------',
    posttitle = '------------------------------------------',
    enter_behavior = "plain",
    edit_behavior = "read",
    bookmarks = [-9,-9,-9,-9,-9,-9],
    tracker = 0,
    seed_func = 'useTwoPartSeed',         # useTwoPartSeed, useThreePartSeed, useAllSeed
    seed_type = 'medieval romance',       # see seed panel for options
    seed_subtype = 'adjective',
    seed_subtype_pretty = 'Desc',
    seed_alt_func = '',                   # set to '' to remove second button
    seed_alt_type = 'medieval romance',
    seed_alt_subtype = 'verb',
    seed_alt_subtype_pretty = 'Action',
    actor_index_state = 0,
    max_character_sheets = 6,
    total_pcs_to_show = 2,
    half_size_rows = 23,
    findList = [],
    findIndex = 0,
    merge = True,
    secrets = [],
    random_chance_list = [0,1,3,5,10,15,20,25,50,75,99,100],
    random_event_chance = 5,
)

formats = dict(
    basefontsize = 14,
    transitory_color = styles.curr_palette['accent2'],    # color3 & ephemeral, format as '9d2e39'
    highlight_color = styles.curr_palette['accent1'],     # color1
    alternate_color = styles.curr_palette['secondary'],   # color2
    # mechanics formatting; expects these keywords but can be swapped around
    oracle = "bold",
    result = "italic",
    query = "bold_italic",
    aside = "italic",
    # don't change these directly; change transitory_color, highlight_color, etc
    mechanic1 = "color1",
    mechanic2 = "color2",
    ephemeral = "color3",
    # fiction formatting; expects these keywords but can be swapped around
    bold = "bold",
    italic = "italic",
    bold_italic = "bold_italic",
    # don't change these directly; change transitory_color, highlight_color, etc
    color1 = "color1",
    color2 = "color2",
    color3 = "color3",
    # scenario/in block links
    link_color = styles.curr_palette['accent2'],
    visited_link_color = styles.curr_palette['accent1'],
)

scenario = dict(
    active = False,
    use_core = True,
    use_oracle = True,
    name = "",
    block = "Start",
    oracle = "fu",
    oracle_func = "fu",
)

user = dict(
    temp = 0,
    # trigger_tests = ['Spot', 'Listen', 'Search']
)

#------------------------------------------------------------------------
# Changes to the following are game-wide and not stored in save games
#------------------------------------------------------------------------

curr_title = "replace me"

# dice presets; first should be 10 items, the second 2
dice_presets = ["1d4", "1d6", "1d8", "1d10", "1d100", "2d4", "2d6", "2d8", "2d10", "1d20"]
dice_spinner_list = ["6", "10"]

# font sizes -- some of these may be deprecated
basefont = str(formats['basefontsize']*.90) + "dp"
basefont60 = str(formats['basefontsize']*.70) + "dp"
basefont75 = str(formats['basefontsize']*.80) + "dp"
basefont80 = str(formats['basefontsize']*.85) + "dp"
basefont90 = str(formats['basefontsize']*.95) + "dp"
baseheight = str(formats['basefontsize']+10) + "dp"
tallheight = str(formats['basefontsize']+15) + "dp"
doubleheight = str(formats['basefontsize']*2+20) + "dp"
tripleheight = str(formats['basefontsize']*3+20) + "dp"
quintupleheight = str(formats['basefontsize']*5+10) + "dp"
octupleheight = str(formats['basefontsize']*8+10) + "dp"
#maintextfont = str(formats['basefontsize']*1) + "dp"
aiheight = baseheight

# font sizes that actually indicate what they do
actortagfont = str(formats['basefontsize']*.90) + "dp"
actorfont = str(formats['basefontsize']*.90) + "dp"
actorstatusfont = str(formats['basefontsize']*.90) + "dp"
blockstatusfont = str(formats['basefontsize']*.90) + "dp"
blockfont = str(formats['basefontsize']*.95) + "dp"
threadstatusfont = str(formats['basefontsize']*.90) + "dp"
threadfont = str(formats['basefontsize']*.90) + "dp"

actortagheight = str((formats['basefontsize']*.90)+20) + "dp"
threadheight = str((formats['basefontsize']*.90)+15) + "dp"

maintextinputfont = str(formats['basefontsize']*.95) + "dp"

fonts = [
    {
        "name": "titlefancyfont",
        "fn_regular": "resources/fonts/miamanueva/miamanueva.ttf",
    },
    {
        "name": "titlefont",
        "fn_regular": "resources/fonts/Cormorant/Cormorant-Regular.ttf",
        "fn_bold": "resources/fonts/Cormorant/Cormorant-Bold.ttf",
        "fn_italic": "resources/fonts/Cormorant/Cormorant-Italic.ttf",
        "fn_bolditalic": "resources/fonts/Cormorant/Cormorant-BoldItalic.ttf"
    },
    {
        "name": "maintextfont",
        "fn_regular": "resources/fonts/Open_Sans/OpenSans-Regular.ttf",
        "fn_bold": "resources/fonts/Open_Sans/OpenSans-Bold.ttf",
        "fn_italic": "resources/fonts/Open_Sans/OpenSans-Italic.ttf",
        "fn_bolditalic": "resources/fonts/Open_Sans/OpenSans-BoldItalic.ttf",
    },
]

#------------------------------------------------------------------------
# Don't mess with anything below this line
#------------------------------------------------------------------------

curr_game_dir = "." + os.sep + "saves" + os.sep + "quicksave" + os.sep
curr_adv = "." + os.sep + "resources" + os.sep + "modules" + os.sep + "moduleA" + os.sep

#------------------------------------------------------------------------
# Config array holders; don't change or edit these
#------------------------------------------------------------------------

actorArray = []
actorStatusArray = []
actorLabelArray = []
actorStatusLabelArray = []

threadArray = []
threadStatusArray = []
threadLabelArray = []
threadStatusLabelArray = []

textArray = []
textStatusArray = []
textLabelArray = []
textFieldLabelArray = []
textStatusLabelArray = []

trackLabelArray = []
trackStatusLabelArray = []

pcKeyLabelArray = []
pcValueLabelArray = []

tempMapArray = []
mapArray = {}

tempGmapArray = []
gmapArray = {}

imgLabelArray = []
#imgTextArray = []


saves/quicksave/config.txt
{"formats": {"mechanic2": "color2", "mechanic1": "color1", "transitory_color": "9d2e39", "bold": "bold", "alternate_color": "ededed", "link_color": "9d2e39", "visited_link_color": "424144", "ephemeral": "color3", "aside": "italic", "basefontsize": 14, "color1": "color1", "color3": "color3", "color2": "color2", "result": "italic", "italic": "italic", "bold_italic": "bold_italic", "oracle": "bold", "query": "bold_italic", "highlight_color": "424144"}, "user": {"temp": 0}, "scenario": {"oracle_func": "fu", "name": "", "use_core": true, "use_oracle": true, "oracle": "fu", "active": false, "block": "Start"}, "general": {"version": "1.1.0", "seed_subtype_pretty": "Seed", "seed_alt_subtype_pretty": "Action", "actor_index_state": 0, "bookmarks": [-9, -9, -9, -9, -9, -9], "posttitle": " ", "random_event_chance": 5, "findIndex": 0, "merge": false, "findList": [], "tracker": 0, "seed_subtype": "adjective", "half_size_rows": 23, "enter_behavior": "plain", "seed_alt_subtype": "verb", "pretitle": " ", "total_pcs_to_show": 2, "secrets": [], "max_character_sheets": 6, "seed_func": "useTwoPartSeed", "seed_type": "medieval romance", "edit_behavior": "read", "seed_alt_func": "", "seed_alt_type": "medieval romance", "random_chance_list": [0, 1, 3, 5, 10, 15, 20, 25, 50, 75, 99, 100]}}

CHANGELOG.md
####Changelog

Please note that this is very rough and just to give you an idea of what's been changed or improved. API breaking and save game breaking stuff may happen!

MAKE BACKUPS BEFORE EDITING FILES MANUALLY.

Version 1.1.0

* overhaul on log files: pick/choose/make your own templates, yaml, cleaner results
* clean up on merge mode (still feels a bit rickety)
* reduced dice preset spinners to two since I'm not finding I use them at all
* added a new section in the footer for dice that need parsing, like ORE
* set random event chance on the fu panel (and set the possible options in config)
* tidied up font formats and heights; now more user customizable
* took out the 'show odds' button on the FU panel; space is at a premium!
* added a new user variable, 'trigger_tests', so you can set your own trigger checks
* added more backup options, including cap on total saves and more options for when to save
* added a new status ("BigQ") to the threads panel for Perilous Intersections compatibility
* added some error catching to the top/bottom jump button
* tweaked a few other generators to give more sensible and usable results
* added a couple of new generators, one for clarifying 'and' and 'but' results, one for character development, both from [Abulafia](http://www.random-generator.com/index.php?title=Main_Page) under CC-BY
* added 'secrets & triggers' panel for adding some surprises to your games
* moved the save for images to the config.user variable, just to eliminate the extra file
* updated help pretty thoroughly
* added a manual edit flag to config to help protect against manual editing overwrites

Version 1.0.0

* added an image panel to the map stack that reads all files out of an 'images' subfolder in a save game and displays them
* combined status and main save files; should load old format fine still too
* seed curation. so much seed curation, so much more needed
* fixed a bug with the actor panel titles; should update correctly now
* new per game setting to determine how many half size (attribute) rows show on a character sheet
* character sheet stack titles will update as soon as the name or nn (nickname) field is exited now
* fixed yet another issue with the collapsing html output; should work now
* tweaked some of the random content on the actor panel to be less gender specific and more broad
* fixed issue where map panels weren't loading pre-existing maps on start
* adjusted the "Generate Grid Room Pattern" some more, added graphical representations
* removed most of the ways to set plain text formatting to streamline interface; use kivy [markup](https://kivy.org/docs/api-kivy.core.text.markup.html) inline if needed

Version 0.9.0

* added in a merge toggle for merging behavior and moved "about" to the title screen
* rewrote the kindom generation on the world panel with new, more useful generators
* wrote up a rough part by part documentation
* more seed curation; there are a LOT of seeds to go through. Recommend using "hardboiled" or "medieval romance" for the most curated results
* hitting "enter" with no input focused will now take a screenshot if debug mode is on
* button for showing the full map and saving a screenshot to your save directory
* fixed up map panels a bit, streamlined saving (no more auto-save)
* new options in the world generator panel for generating grid patterns (rough)
* added a simple grid map panel for grid-based dungeon crawls
* merge mode; choose if your save game consolidates sequential identical tags or not (or vice versa)
* dice presets cleaned up, now customizable in config
* buttons now have a border for better usability
* as always, bug fixes

Version 0.8.0

* formatting now pulls from the config file and can be set on a tag by tag basis
* refactored some things to help avoid identification issues when two lines of block text were identical
* added a toggle for a basic "debug mode" to config; with it False you will see a lot less messaging, with True, a lot more (and hopefully more helpful) messaging. If your game is working correctly, ignore the messaging
* fixed scrolling and jumps and bookmarks so hopefully they work correctly now
* basic "Find" and "Next" functions for quickly jumping around (still experimental).
* deprecated "don't show" in favor of "ephemeral"; ephemeral now triggers the old "don't show" behavior
* renamed 'clean' to 'fiction' and 'cedit' to 'fic-edit' (for now)
* separated out block formatting into clear "fiction" and "mechanics" categories; the appearance is the same but "mechanics" tags won't show in "fiction" editing modes or in fiction-only logs
* added support to logs for some markdown tags if placed in blocks (italic, bold, superscript, subscript, underline, strikethrough). Underline and strikethrough won't display in block but the markdown and html logs will appear correctly
* moved play logs to a sub directory and added some "no mechanics" versions
* updated PC panel to support multiple PCs; renamed to "Character Sheets"
* pulled out unused stuff from pythia.py
* updated post and pretitle to textfields and cleared config issue carrying them over
* "resolved" status for a thread will now send it to the bottom of the list
* seeds are much better curated (but there's still a lot of them so I might have missed something)

Version 0.7.1

* fixed a couple of issues related to refactoring the main display routine, mainly formatting updates on status toggle and editing text carrying over to every mode. Looks great now!
* a few other bug fixes
* removed some old commented out code

Version 0.7.0

* refactored main display routine so switching modes should work much much faster now.
* actor index to make navigating long actor rosters easier
* "don't show" status; items with this tag will no longer show up in main window. Ever.
* added a new but/and/- oracle weighted towards lots of events
* added two new Simple World (Apocalypse World) inspired generation buttons for actor/plot moves
* reversed panel adding order

config.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-
##---------------------------------------------------------------------------------------
#  Configuration and variables
#
##---------------------------------------------------------------------------------------

import styles
import os

# set this to True to get lots of messaging
debug = True

# set this to true if you are manually editing your main file
# then back to False when you're satisfied it loads properly
# game will NOT save if this is True
manual_edit_mode = False

# this controls how often full zip backups are made
# options are 'app_exit' and 'app_start', and you can use both
backup_behavior = ['app_exit']
# backup_limit should be 0 (for unlimited) or a positive number
# if set to a negative, no backups will be made
backup_limit = 7

#------------------------------------------------------------------------
# Changes to the following are stored in save games
#------------------------------------------------------------------------

general = dict(
    version = '1.1.0',
    pretitle = '------------------------------------------',
    posttitle = '------------------------------------------',
    enter_behavior = "plain",
    edit_behavior = "read",
    bookmarks = [-9,-9,-9,-9,-9,-9],
    tracker = 0,
    seed_func = 'useTwoPartSeed',         # useTwoPartSeed, useThreePartSeed, useAllSeed
    seed_type = 'medieval romance',       # see seed panel for options
    seed_subtype = 'adjective',
    seed_subtype_pretty = 'Desc',
    seed_alt_func = '',                   # set to '' to remove second button
    seed_alt_type = 'medieval romance',
    seed_alt_subtype = 'verb',
    seed_alt_subtype_pretty = 'Action',
    actor_index_state = 0,
    max_character_sheets = 6,
    total_pcs_to_show = 2,
    half_size_rows = 23,
    findList = [],
    findIndex = 0,
    merge = False,
    secrets = [],
    random_chance_list = [0,1,3,5,10,15,20,25,50,75,99,100],
    random_event_chance = 5,
)

formats = dict(
    basefontsize = 14,
    transitory_color = styles.curr_palette['accent2'],    # color3 & ephemeral, format as '9d2e39'
    highlight_color = styles.curr_palette['accent1'],     # color1
    alternate_color = styles.curr_palette['secondary'],   # color2
    # mechanics formatting; expects these keywords but can be swapped around
    oracle = "bold",
    result = "italic",
    query = "bold_italic",
    aside = "italic",
    # don't change these directly; change transitory_color, highlight_color, etc
    mechanic1 = "color1",
    mechanic2 = "color2",
    ephemeral = "color3",
    # fiction formatting; expects these keywords but can be swapped around
    bold = "bold",
    italic = "italic",
    bold_italic = "bold_italic",
    # don't change these directly; change transitory_color, highlight_color, etc
    color1 = "color1",
    color2 = "color2",
    color3 = "color3",
    # scenario/in block links
    link_color = styles.curr_palette['accent2'],
    visited_link_color = styles.curr_palette['accent1'],
)

scenario = dict(
    active = False,
    use_core = True,
    use_oracle = True,
    name = "",
    block = "Start",
    oracle = "fu",
    oracle_func = "fu",
)

user = dict(
    temp = 0,
    # trigger_tests = ['Spot', 'Listen', 'Search']
)

#------------------------------------------------------------------------
# Changes to the following are game-wide and not stored in save games
#------------------------------------------------------------------------

# do you want spinners or cycling text buttons?
use_spinner_status = True

# if this is true the game name will be used for the title in logs
use_autotitle_in_logs = True

# this will be copied in as a YAML into log templates calling for it
# 'title' is also used for html play logs
yaml = dict(
    title = "Play Log",
    author = "your name",
    mainfont = "Lora",
    sansfont = "Ubuntu Mono",
    monofont = "Ubuntu Mono",
    fontsize = "10pt",
)

# dice presets; first should be 10 items, the second 2
dice_presets = ["1d4", "1d6", "1d8", "1d10", "1d100", "2d4", "2d6", "2d8", "2d10", "1d20"]
dice_spinner_list = ["6", "10"]

# font sizes -- some of these may be deprecated
basefont = str(formats['basefontsize']*.90) + "dp"
basefont60 = str(formats['basefontsize']*.70) + "dp"
basefont75 = str(formats['basefontsize']*.80) + "dp"
basefont80 = str(formats['basefontsize']*.85) + "dp"
basefont90 = str(formats['basefontsize']*.95) + "dp"
baseheight = str(formats['basefontsize']+10) + "dp"
tallheight = str(formats['basefontsize']+15) + "dp"
doubleheight = str(formats['basefontsize']*2+20) + "dp"
tripleheight = str(formats['basefontsize']*3+20) + "dp"
quintupleheight = str(formats['basefontsize']*5+10) + "dp"
octupleheight = str(formats['basefontsize']*8+10) + "dp"
#maintextfont = str(formats['basefontsize']*1) + "dp"
aiheight = baseheight

# font sizes that actually indicate what they do
actortagfont = str(formats['basefontsize']*.90) + "dp"
actorfont = str(formats['basefontsize']*.90) + "dp"
actorstatusfont = str(formats['basefontsize']*.90) + "dp"
blockstatusfont = str(formats['basefontsize']*.90) + "dp"
blockfont = str(formats['basefontsize']*.95) + "dp"
threadstatusfont = str(formats['basefontsize']*.90) + "dp"
threadfont = str(formats['basefontsize']*.90) + "dp"

actortagheight = str((formats['basefontsize']*.90)+20) + "dp"
threadheight = str((formats['basefontsize']*.90)+15) + "dp"

maintextinputfont = str(formats['basefontsize']*.95) + "dp"

fonts = [
    {
        "name": "titlefancyfont",
        "fn_regular": "resources/fonts/miamanueva/miamanueva.ttf",
    },
    {
        "name": "titlefont",
        "fn_regular": "resources/fonts/Cormorant/Cormorant-Regular.ttf",
        "fn_bold": "resources/fonts/Cormorant/Cormorant-Bold.ttf",
        "fn_italic": "resources/fonts/Cormorant/Cormorant-Italic.ttf",
        "fn_bolditalic": "resources/fonts/Cormorant/Cormorant-BoldItalic.ttf"
    },
    {
        "name": "maintextfont",
        "fn_regular": "resources/fonts/Open_Sans/OpenSans-Regular.ttf",
        "fn_bold": "resources/fonts/Open_Sans/OpenSans-Bold.ttf",
        "fn_italic": "resources/fonts/Open_Sans/OpenSans-Italic.ttf",
        "fn_bolditalic": "resources/fonts/Open_Sans/OpenSans-BoldItalic.ttf",
    },
]

#------------------------------------------------------------------------
# Don't mess with anything below this line
#------------------------------------------------------------------------

curr_game_dir = "." + os.sep + "saves" + os.sep + "quicksave" + os.sep
curr_adv = "." + os.sep + "resources" + os.sep + "modules" + os.sep + "moduleA" + os.sep

#------------------------------------------------------------------------
# Config array holders; don't change or edit these
#------------------------------------------------------------------------

actorArray = []
actorStatusArray = []
actorLabelArray = []
actorStatusLabelArray = []

threadArray = []
threadStatusArray = []
threadLabelArray = []
threadStatusLabelArray = []

textArray = []
textStatusArray = []
textLabelArray = []
textFieldLabelArray = []
textStatusLabelArray = []

trackLabelArray = []
trackStatusLabelArray = []

pcKeyLabelArray = []
pcValueLabelArray = []

tempMapArray = []
mapArray = {}

tempGmapArray = []
gmapArray = {}

imgLabelArray = []
#imgTextArray = []
